### PR TITLE
types/data: the bit data type should be treat as uint64 (#11951)

### PR DESCRIPTION
**Pull request by gaoxingliang**
### What problem does this PR solve? 
fix cherry-pick #11951 to release-2.1 .  related issue for automatically pick is in [here](https://github.com/pingcap/tidb/pull/12424#event-2672053733)

### CODE REVIEW

File: executor/insert_test.go [hunk 1]
```diff
@@ -471,3 +472,21 @@ func (s *testSuite) TestPartitionInsertOnDuplicate(c *C) {
 	tk.MustQuery(`select * from t2`).Check(testkit.Rows("1 1"))
 
 }
+
+func (s *testSuite) TestBitInRange(c *C) {
```
Comment (SunRunAway) [inline executor/insert_test.go]: in #11951, it is callled `TestBit`, why here is it called `TetBItInRange`

**Comment by SunRunAway** [comment]
Should we abandon #12424 ? 
And what's the difference this PR from #12424?

**Comment by gaoxingliang** [comment]
Hi @SunRunAway   I created this PR because the cherry pick failed because of code conflicts and that's why the method name changed. please see the [PR](https://github.com/pingcap/tidb/pull/12424) comments and[ the error log ](https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/tidb_ghpr_unit_test/runs/10004/nodes/78/steps/297/log/?start=0) it shows build error.

> [2019-10-13T15:59:04.620Z] # github.com/pingcap/tidb/executor_test [github.com/pingcap/tidb/executor.test]
[2019-10-13T15:59:04.620Z] executor/insert_test.go:492:10: undefined: testSuite3
[2019-10-13T15:59:04.620Z] FAIL	github.com/pingcap/tidb/executor [build failed]

**Comment by SunRunAway** [comment]
Hi, I've resolved the conflicts in #12424, and will close this PR.
Thank you anyway, @gaoxingliang

**Pull request closed without merging**